vyper/builtins/functions.py

```diff
@@ -294,6 +294,48 @@ class Slice(BuiltinFunction):
     ]
     _return_type = None
 
+    def evaluate(self, node):
+        validate_call_args(node, 3)
+        bytestring, start_node, length_node = node.args
+        if not (
+            isinstance(bytestring, (vy_ast.Bytes, vy_ast.Str, vy_ast.Hex))
+            and isinstance(start_node, vy_ast.Int)
+            and isinstance(length_node, vy_ast.Int)
+        ):
+            raise UnfoldableNode
+
+        (start, length) = (start_node.value, length_node.value)
+
+        if start < 0:
+            raise ArgumentException("Start cannot be negative", start_node)
+        elif length <= 0:
+            raise ArgumentException("Length must be positive", length_node)
+
+        if isinstance(bytestring, vy_ast.Hex):
+            bytes_value = bytes.fromhex(bytestring.value.removeprefix("0x"))
+            if start >= 32:
+                raise ArgumentException("Start cannot take that value", start_node)
+            if length > 32:
+                raise ArgumentException("Length cannot take that value", length_node)
+            if len(bytes_value) != 32:
+                raise ArgumentException("Length can only be of 32", bytestring)
+        elif isinstance(bytestring, vy_ast.Str):
+            bytes_value = bytestring.value
```

Comment:
@charles-cooper how do you feel about this line ? Here the `bytes_value` type is `str` while otherwise it's `bytes`

Reply:
looks confusing